src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -61,6 +61,12 @@ public HttpContext Context
         /// <inheritdoc />
         public bool IsPartial { get; set; }
 
+        /// <summary>
+        /// Gets the <see cref="IHtmlEncoder"/> to be used for encoding HTML.
+        /// </summary>
+        [Activate]
+        public IHtmlEncoder HtmlEncoder { get; set; }
```

Comment:
@NTaylorMullen the issue with auto-injection is references to the injected properties (e.g. `Html`) don't compile.  So the `WriteTo` @ajaybhargavb changed in the next difference would cause build failures.